### PR TITLE
Feature/grpc to rest migration

### PR DESCRIPTION
**Pull request by Azher2Ali**
Committing changes with an updated PR for #400 and #401. The issues related to swagger dev link and authorization for valid JWT and 401= no token or invalid token is completed.

### CODE REVIEW

File: src/main/java/org/icgc/argo/program_service/security/EgoRestSecurity.java [hunk 1]
```diff
@@ -40,7 +41,7 @@
 
 @Slf4j
 @Service
-@Profile("auth")
+@Profile("default")
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/security/EgoRestSecurity.java:44]: This should still only run for `auth` profile

File: src/main/java/org/icgc/argo/program_service/security/EgoRestSecurity.java [hunk 2]
```diff
@@ -57,11 +58,16 @@ public Optional<EgoToken> verifyRestTokenHeader(String jwtToken) {
...
+        val jwt = verifier.verify(jwtToken);
+        return parseToken(jwt);
+      } else {
+        log.warn("No Token or Invalid Token");
+        throw new UnauthorizedException("No Token or Invalid Token");
+      }
     } catch (JWTVerificationException | NullPointerException e) {
       log.warn(e.getMessage());
-      throw new NotFoundException("No Token or Invalid Token");
+      throw new UnauthorizedException("No Token or Invalid Token");
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/security/EgoRestSecurity.java:70]: 👍

File: src/main/java/org/icgc/argo/program_service/security/JWTAuthorizationFilter.java [hunk 3]
```diff
@@ -27,23 +27,30 @@
...
 import org.springframework.context.annotation.Profile;
 import org.springframework.http.HttpHeaders;
 import org.springframework.stereotype.Component;
 import org.springframework.web.filter.OncePerRequestFilter;
+import org.springframework.web.servlet.HandlerExceptionResolver;
 
 @Slf4j
 @Component
-@Profile("auth")
+@Profile("default")
```
Comment (joneubank) [inline src/main/java/org/icgc/argo/program_service/security/JWTAuthorizationFilter.java:43]: This should remain as requiring `auth` profile

**Review by joneubank — CHANGES_REQUESTED** [review]
Should restore the auth handlers to only be applied when we have `auth` profile.

File: src/main/java/org/icgc/argo/program_service/security/JWTAuthorizationFilter.java (hunk 3 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/security/JWTAuthorizationFilter.java]: Changed to auth

File: src/main/java/org/icgc/argo/program_service/security/EgoRestSecurity.java (hunk 1 above)
Comment (Azher2Ali) [reply src/main/java/org/icgc/argo/program_service/security/EgoRestSecurity.java]: changed profile to auth

**Review by joneubank — APPROVED** [review]

**Pull request merged**